security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -260,7 +263,15 @@ func (s *sdsservice) StreamSecrets(stream sds.SecretDiscoveryService_StreamSecre
 			defer recycleConnection(conID, resourceName)
 
 			conIDresourceNamePrefix := sdsLogPrefix(conID, resourceName)
-			if !s.skipToken {
+			if s.localJWT {
+				// Running in-process, no need to pass the token from envoy to agent as in-context - use the file
```

Comment:
I am uncertain why the localJWT option is needed. Without the localJWT option, Envoy reads the token from a file and passes it to istio agent in the context. Withou localJWT option, istio agent reads the token from a file.
If the localJWT option is introduced to improve the performance, as the token still needs to be read, there is no performance gain.

Reply:
This is to simplify the code - there is no need for Pilot to generate a complicated config, with the file path - in each cluster/listener - when the agent has access to the file as well.

The option is only used short term, until node-agent is still in use - once it's gone there will be no passing of tokens from envoy to SDS.

Reply:
As a fun fact - if the file is not found, envoy is crashing hard. 

Not passing the file from envoy is not only simpler - it also allows us the option to use Envoy GRPC library for envoy to local SDS communication, instead of using 2 grpc stacks at the same time.

Reply:
Be a good citizen -- file an issue about the crash?

Reply:
I think John did already, he found it while testing istiod.

Reply:
Can you provide more details about "using 2 grpc stacks at the same time"?

Reply:
Opened https://github.com/envoyproxy/envoy/issues/8941